app/models/concerns/fahrplan_updater.rb

```diff
@@ -3,20 +3,26 @@ module FahrplanUpdater
   include FahrplanParser
 
   def fill_event_info
-    if conference.downloaded?
-      fahrplan = FahrplanParser.new(self.conference.schedule_xml)
-      info = fahrplan.event_info_by_guid[self.guid]
-      return if info.empty?
-      update_event_info(info)
-    end
+    return unless conference.downloaded?
+    fahrplan = FahrplanParser.new(conference.schedule_xml)
+    info = fahrplan.event_info_by_guid[guid]
+    return if info.empty?
+    update_event_info(info)
   end
 
   # update event attributes from schedule XML
   def update_event_info(info)
     self.title = info.delete(:title)
     id = info.delete(:id)
-    self.link = self.conference.get_event_url(id)
-    info[:tags] << id.to_s
-    self.update_attributes info
+    self.metadata[:remote_id] = id
+    self.link = get_event_url(conference, id)
+    update_attributes info
+  end
+
+  private
+
+  def get_event_url(conference, id)
+    return unless conference.schedule_url.present?
+    conference.schedule_url.sub('schedule.xml', "events/#{id}.html").freeze
```

Comment:
Why did you move this getter fro mthe conference-model, as it seems to require the conference anyway (now as an argument)

Also, this logic will only work for schedules.xmls directly generated by a frab. Often we build our own schedule.xml with scripts on a c3voc-host, where this assumption does not hold, but as you didn't change this in this PR this might be the wrong place to change that.